libvast/vast/concept/parseable/vast/data.hpp

```diff
@@ -48,9 +49,13 @@ struct data_parser : parser<data_parser> {
     auto ws = ignore(*parsers::space);
     auto x = ws >> ref(p) >> ws;
     auto kvp = x >> "->" >> x;
-    auto field = parsers::identifier >> ":" >> x;
+    auto field = ws >> parsers::identifier >> ":" >> x;
+    auto anonymous = detail::inject(std::string{}) >> x;
```

Comment:
Instead of writing a new helper utility, I would rely on semantic actions to achieve your desired goal:

```cpp
auto to_field = [](std::string value) {
  return std::make_pair(std::string{}, data{std::move(value)});
};
auto anon_field = x ->* to_field;
```

And then ditch the `inject` craziness. 😉